modopt/math/metrics.py

```diff
@@ -10,12 +10,24 @@
 
 import numpy as np
 from scipy.ndimage import uniform_filter, gaussian_filter
-try:
-    from skimage.metrics import structural_similarity as compare_ssim
-except ImportError:  # pragma: no cover
-    import_skimage = False
-else:
-    import_skimage = True
+import sys
+
+if sys.version_info.major == 3:
+    if sys.version_info.minor == 5:
+        try:
+            from skimage.measure import compare_ssim
+        except ImportError:  # pragma: no cover
+            import_skimage = False
+        else:
+            import_skimage = True
+
+    elif sys.version_info.minor > 5:
+        try:
+            from skimage.metrics import structural_similarity as compare_ssim
+        except ImportError:  # pragma: no cover
+            import_skimage = False
+        else:
+            import_skimage = True
```

Comment:
Since we longer support python versions <3.5 perhaps something like this would be a bit cleaner:

```python
try:
    if sys.version_info.minor == 5:
        from skimage.measure import compare_ssim
    elif sys.version_info.minor > 5:
        from skimage.metrics import structural_similarity as compare_ssim
except ImportError:  # pragma: no cover
    import_skimage = False
else:
    import_skimage = True
```

Reply:
Sorry to be a pain, but the point of my suggestion was to avoid having two try/except statements, where as far as I can see we only need to add an if statement inside the try statement.

Reply:
Yes sure